pkgs/development/python-modules/kornia-rs/default.nix

```diff
@@ -0,0 +1,63 @@
+{ lib
+, fetchFromGitHub
+, buildPythonPackage
+, rustPlatform
+, cmake
+, nasm
+}:
+
+buildPythonPackage rec {
+  pname = "kornia-rs";
+  version = "0.1.2";
+
+  src = fetchFromGitHub {
+    owner = "kornia";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-7toCMaHzFAzm6gThVLBxKLgQVgFJatdJseDlfdeS8RE=";
+  };
+
+  format = "pyproject";
+
+  nativeBuildInputs = [
+    rustPlatform.maturinBuildHook
+    rustPlatform.cargoSetupHook
+    cmake # Only for dependencies.
+    nasm # Only for dependencies.
+  ];
+
+  cargoRoot = "py-kornia";
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
+
+  prePatch = ''
+    # The path dependency doesn't vendor the dependencies correctly, so get kornia-rs from crates instead.
+    patch -p1 <<END_PATCH
+    --- a/py-kornia/Cargo.toml
+    +++ b/py-kornia/Cargo.toml
+    @@ -9,6 +9,6 @@ crate-type = ["cdylib"]
+
+     [dependencies]
+     anyhow = "1.0.80"
+    -kornia-rs = { path = ".." }
+    +kornia-rs = { version = "${version}" }
+     pyo3 = { version = "0.20", features = ["extension-module"] }
+     numpy = { version = "0.20.0" }
+
+    END_PATCH
+
+    cp ${./Cargo.lock} py-kornia/Cargo.lock
```

Comment:
Can't we patch the upstream one instead?

Reply:
That one is different though because they use the path dependency.